README

                             Compress-Zlib

                             Version 2.014

                            2nd September 2008

       Copyright (c) 1995-2008 Paul Marquess. All rights reserved.
          This program is free software; you can redistribute it
           and/or modify it under the same terms as Perl itself.

DESCRIPTION
-----------

This module provides a Perl interface to the zlib compression library.  

PREREQUISITES
-------------

Before you can build Compress-Zlib you need to have the following
installed on your system:

    * Perl 5.004 or better. 
    * Compress::Raw::Zlib
    * IO::Compress::Gzip

BUILDING THE MODULE
-------------------

Assuming you have met all the prerequisites, the module can now be built
using this sequence of commands:

    perl Makefile.PL
    make
    make test

INSTALLATION
------------

To install Compress-Zlib, run the command below:

    make install

TROUBLESHOOTING
---------------

Undefined Symbol gzsetparams
----------------------------

If you get the error shown below when you run the Compress-Zlib test
harness it probably means you are running a copy of zlib that is
version 1.0.5 or older.

t/01version.........Can't load 'blib/arch/auto/Compress/Zlib/Zlib.so' for 
                    module Compress::Raw::Zlib: blib/arch/auto/Compress/Raw/Zlib/Zlib.so:
                    undefined symbol: gzsetparams at ...

There are two ways to fix this problem:

    1. Upgrade to the latest version of zlib.

    2. Edit config.in and set the OLD_ZLIB variable to True.

Test Harness 01version fails
----------------------------
If the 01version test harness fails, and the problem isn't covered by the
scenario above, it probably means that you have two versions of 
zlib installed on your system.

Run the command below to see if this is indeed the case

    make test TEST_VERBOSE=1 TEST_FILES=t/01version.t 

Try removing the one you don't want to use and rebuild.   

FEEDBACK
--------

How to report a problem with Compress-Zlib.

To help me help you, I need all of the following information:

 1. The Versions of everything relevant.
    This includes:

     a. The *complete* output from running this
     
            perl -V
     
        Do not edit the output in any way.
        Note, I want you to run "perl -V" and NOT "perl -v".
     
        If your perl does not understand the "-V" option it is too
        old. This module needs Perl version 5.004 or better.     

     b. The version of Compress-Zlib you have. 
        If you have successfully installed Compress-Zlib, this one-liner
        will tell you:

           perl -MCompress::Zlib -e 'print qq[ver $Compress::Zlib::VERSION\n]'

        If you are  running windows use this

           perl -MCompress::Zlib -e "print qq[ver $Compress::Zlib::VERSION\n]"

        If you haven't installed Compress-Zlib then search Compress::Zlib.pm
        for a line like this:

          $VERSION = "2.014" ;

 2. If you are having problems building Compress-Zlib, send me a
    complete log of what happened. Start by unpacking the Compress-Zlib
    module into a fresh directory and keep a log of all the steps

        [edit config.in, if necessary]
        perl Makefile.PL
        make
        make test TEST_VERBOSE=1          

Paul Marquess <pmqs@cpan.org>


Changes
CHANGES
-------

  2.014 2 September 2008

      * Makefile.PL
        Updated to check for indirect dependencies.

  2.012 15 July 2008

      * No Changes

  2.011 17 May 2008

      * No Changes

  2.010 5 May 2008

      * Fixed problem that meant Perl 5.10 could not upgrade this module.
        [RT #35341]

  2.009 20 April 2008

      * Minor documentation issue with flush.
        [rt.cpan.org #31446]

  2.008 2 November 2007

      * Minor documentation changes in README

  2.007 22 September 20007

      * lib/Compress/Zlib.pm -- 1.x Backward Compatability issues
        gzclose - documented return value was wrong, should be 0 for ok.
        gzflush - return value didn't match 1.x, should return 0 if ok.
        [rt.cpan.org #29215] and Debian bug #440943 http://bugs.debian.org/440943

  2.006 1 September 20007

      * Makefile.PL
        Added INSTALLDIRS directive to install as a core module when built
        on a perl >= 5.9.

      * t/03zlib-v1.t
        Fixed crc32 and adler32 tests in to remove ascii assumption.

      * lib/Compress/Zlib.pm
        Make gzreadline not support $/, just like in Compress::Zlib 1.x
        Folk who want $/ with readline support can get it in 
        IO::Uncompress::Gunzip.
        [rt.cpan.org #28663] and Debian bug #435656 http://bugs.debian.org/435656
  2.005 18 June 2007

      * Added info about removing Compress::Zlib version 1, before
        installing version 2. 

  2.004 3 March 2007

      * rewrote memGzip using IO::Compress::Gzip::gzip

  2.003 2 January 2007

      * Added explicit version checking

  2.002 29 December 2006

      * Documentation updates.

      * Fix append mode with gzopen.
        rt-cpan.org 24041

      * Allow gzopen to read from and write to a scalar reference.

  2.001 1 November 2006

      * Remove beta status.

  2.000_14 26 October 2006

      * No changes.

  2.000_13 20 June 2006

      * No changes.

  2.000_12 16 April 2006

      * Fixed gzread to zap the output buffer to an empty string when zero
        bytes are requested. This matches the behaviour of C::Z 1.x

  2.000_11 10 April 2006

      * No changes.

  2.000_10 13 March 2006

      * Changed gzread so that its behaviour matches C::Z::gzread 1.x if it
        is called after eof. In this case it will write an empty string
        into the output parameter. This change is solely for backward
        compatability reasons.

  2.000_09 3 March 2006

      * Posted to CPAN

  2.000_08 2 March 2006

      * Breakout zlib specific code into separate modules.

      * Limited support for reading/writing zip files

  2.000_06 5 October 2005

      * Added eof parameter to Compress::Zlib::inflate method.

      * Fixed issue with 64-bit

  2.000_05 4 October 2005

      * Renamed IO::* to IO::Compress::* & IO::Uncompress::*

  2.000_04 23 September 2005

      * Fixed some more non-portable test that were failing on VMS.

      * fixed problem where error messages in the oneshot interface were
        getting lost.

  2.000_03 12 September 2005

      * Fixed some non-portable test that were failing on VMS.

      * Fixed export of zlib constants from the IO::* classes

  2.000_02 6 September 2005

      * Split Append mode into Append and Merge

      * Fixed typos in the documentation.

      * Added pod/FAQ.pod

      * Added libscan to Makefile.PL

      * Added InputLength for IO::Gunzip et al

  2.000_01 22 August 2005

      * Fixed VERSION in Compress::Gzip::Constants

      * Removed Compress::Gzip::Info from the distribution.

  2.000_00 21 August 2005

      * First Beta relase of Compress::zlib rewrite.

  1.38 - 6 September 2005

      * Integrate core change 25304 -- Symbian Update

      * Added libscan to Makefile.PL

  1.37 - 12 August 2005

      * Change to t/03examples.t for VMS from Abe Timmerman

  1.36 - 3 August 2005

      * Renamed zlib-src-1.2.3 to zlib-src to help VMS

      * Fixed Makefile.PL for VMS

      * Fixed t/03examples.t for VMS

      * Added a couple of notes about incompatibility with Unix compress.

  1.35 - 16 July 2005

      * Updated zlib source to 1.2.3

      * Fixed problem with where two calls to gzclose would hang the debugger.
        See https://rt.cpan.org/Ticket/Display.html?id=13789

      * Added code from Alexey Tourbin to use XSLoader when available,
        and DynaLoader otherwise.

      * Documented that the compress & uncompress functions were not
        the same as the Unix utilities of the same name.

      * Fixed 05gzsetp -- it left a temp file hanging around. 

      * Integrate core change 24787 - SvUPGRADE returns void in blead

      * Integrate core change 24788 - Makefile.PL adjustments for the core


  1.34 - 30 January 2005

      * Fixed typo in the README

      * Fixed examples.t on Win32 where paths have embedded whitespace.

      * Fix for Cygwin and core integration from Jos I. Boumans

      * Upgrade zlib source to 1.2.2

  1.33 - 14 January 2004

      * Reworked Makefile.PL to avoid creating a private copy of zlib. 
        This both simplifies the build, plus it makes life easier for VMS.

      * Patches for Makefile.PL to get it to work on VMS supplied by
        Craig A. Berry.

      * memGunzip has very slow on FreeBSD. Turns out to be down to
        the way realloc works on FreeBSD. Changed both inflate & deflate
        to use exponentially increasing buffer sizes when they need to
        realloc. Thanks to Peter Jeremy for the lowdown on FreeBSD
        memory allocation.

  1.32 - 26 November 2003

      * Steve Hay reported a problem on rt.cpan.org with Windows and
        MSCV++ 6.0 where the source from the zlib directory was getting
        installed with the rest of the module.
        https://rt.cpan.org/Ticket/Display.html?id=1741

        This has been fixed by renaming the "zlib" directory to "zlib-src"
        thus avoiding a conflict with the name of this Perl module.


Makefile.PL
#! perl -w

use strict ;
require 5.004 ;

$::VERSION = '2.014' ;

use private::MakeUtil;
use ExtUtils::MakeMaker 5.16 ;


UpDowngrade(getPerlFiles('MANIFEST')) 
    unless $ENV{PERL_CORE};

# Modules supplied by this distribution    
my @myModules = qw( Compress::Zlib);

# Direct dependencies                  
my @MainPREREQ = qw(Compress::Raw::Zlib
                    IO::Compress::Base
                    IO::Compress::Base::Constants
                    IO::Uncompress::Base
                    IO::Compress::Gzip
                    IO::Compress::Gzip::Constants
                    IO::Uncompress::Gunzip
                   );

# Get indirect dependencies that will break if we install this version
push @MainPREREQ, FindBrokenDependencies($::VERSION, @myModules)
    unless $ENV{PERL_CORE} ;                   
    
my %MainPREREQ = map { $_ => $::VERSION } @MainPREREQ; 

WriteMakefile( 
    NAME         => 'Compress::Zlib',
    VERSION_FROM => 'lib/Compress/Zlib.pm',
    'dist'       => { COMPRESS     => 'gzip', 
                      TARFLAGS     => '-chvf',
                      SUFFIX       => 'gz',
                      DIST_DEFAULT => 'MyTrebleCheck tardist',
                    },

    (
      $ENV{SKIP_FOR_CORE}
        ? (MAN3PODS    => {})
	    : (PREREQ_PM   => { %MainPREREQ,
		                    'Scalar::Util'                  => 0,
                          }
	      )
    ),
       
    (
    $] >= 5.005
        ? (ABSTRACT_FROM => 'lib/Compress/Zlib.pm',
            AUTHOR       => 'Paul Marquess <pmqs@cpan.org>')
        : ()
    ),

    INSTALLDIRS => ($] >= 5.009 ? 'perl' : 'site'),

    (
    $] >= 5.009 && ! $ENV{PERL_CORE}
        ? (INST_LIB    => 'blib/arch')
        : ()
    ),

    ((ExtUtils::MakeMaker->VERSION() gt '6.30') ?
        ('LICENSE'  => 'perl')         : ()),    

) ;

# Check for Compress::Zlib version 1.
eval  "require Compress::Zlib; " ;
if ( ! $ENV{PERL_CORE} && ! $@ && $Compress::Zlib::VERSION < 2)
{
    my $ver = $Compress::Zlib::VERSION ;
    print <<EOM

I see you already have Compress::Zlib version $ver installed.
Version 1 MUST be removed for version 2 to operate correctly.

The easiest way to do that is to install Compress::Zlib like this

    make install UNINST=1

If you run the CPAN shell run this before installing Compress::Zlib

    o conf make_install_arg UNINST=1

EOM
}

# end of file Makefile.PL

